op-service/dial/active_l2_provider_test.go

```diff
@@ -0,0 +1,104 @@
+package dial
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestActiveSequencerFailoverBehavior_RollupProvider tests that the ActiveL2RollupProvider
+// will failover to the next provider if the current one is not active.
+func TestActiveSequencerFailoverBehavior_RollupProviders(t *testing.T) {
+	// Create two mock rollup clients, one of which will declare itself inactive after first check.
+	primarySequencer := testutils.MockRollupClient{}
```

Comment:
maybe just do
```suggestion
	primarySequencer := new(testutils.MockRollupClient)
```
so you can naturally treat them as pointers down below and don't have to always get the reference.